debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MysqlConnectorSslIT.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.math.BigDecimal;
+import java.net.URL;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.config.Configuration;
+import io.debezium.doc.FixFor;
+import io.debezium.embedded.AbstractConnectorTest;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.util.Testing;
+
+public class MysqlConnectorSslIT extends AbstractConnectorTest {
+    private static final String TABLE_NAME = "dbz4787";
+    private static final String TRUST_STORE_PATH = "ssl/truststore";
+    private static final String TRUST_STORE_PASSWORD = "debezium";
+    private static final Path DB_HISTORY_PATH = Testing.Files.createTestingPath("file-db-history-decimal.txt")
+            .toAbsolutePath();
+    private final UniqueDatabase DATABASE = new UniqueDatabase("ssldb", "ssl_test")
+            .withDbHistoryPath(DB_HISTORY_PATH);
+
+    private Configuration config;
+
+    @Before
+    public void beforeEach() {
+        stopConnector();
+        DATABASE.createAndInitialize();
+        initializeConnectorTestFramework();
+        Testing.Files.delete(DB_HISTORY_PATH);
+    }
+
+    @After
+    public void afterEach() {
+        try {
+            stopConnector();
+        }
+        finally {
+            Testing.Files.delete(DB_HISTORY_PATH);
+        }
+    }
+
+    @Test
+    @FixFor("DBZ-4787")
+    public void testSslRequireModeWithNullStore() throws SQLException, InterruptedException {
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.SCHEMA_ONLY)
+                .with(MySqlConnectorConfig.TABLE_INCLUDE_LIST, DATABASE.qualifiedTableName(TABLE_NAME))
+                .with(MySqlConnectorConfig.SSL_MODE, MySqlConnectorConfig.SecureConnectionMode.REQUIRED)
+                .build();
+
+        start(MySqlConnector.class, config);
+
+        waitForSnapshotToBeCompleted("mysql", DATABASE.getServerName());
+        insertOneRecord();
+
+        assertChangeRecord(consumeInsert());
+
+        stopConnector();
+    }
+
+    @Test
+    @FixFor("DBZ-4787")
+    public void testSslRequireMode() throws SQLException, InterruptedException {
+        final URL trustStoreFile = MysqlConnectorSslIT.class.getClassLoader().getResource(TRUST_STORE_PATH);
```

Comment:
Where is this file coming from? I don't see it added as part of this PR? I.e. how is this test passing on CI? I feel like I'm still missing some detail.